presto-tablestore/src/main/java/com/facebook/presto/tablestore/TablestoreAbstractRecordCursor.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tablestore;
+
+import com.alicloud.openservices.tablestore.ClientException;
+import com.alicloud.openservices.tablestore.TableStoreException;
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.RecordCursor;
+
+import java.util.Base64;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static com.facebook.presto.tablestore.TablestoreFacade.NANO_SECS_IN_ONE_MS;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.System.currentTimeMillis;
+import static java.lang.System.nanoTime;
+import static java.util.Objects.requireNonNull;
+
+public abstract class TablestoreAbstractRecordCursor
+        implements RecordCursor
+{
+    private static final Logger log = Logger.get(TablestoreAbstractRecordCursor.class);
+    private static final long NANOS_PER_MILLI_SECOND = 1000_000;
+
+    protected interface Callable<T>
+    {
+        T call();
+    }
+
+    protected final TablestoreColumnHandle[] columnHandles;
+    protected final List<TablestoreColumnHandle> primaryKeys;
+    protected final long startTimestamp;
+    protected final String tableName;
+    protected final ConnectorSession session;
+    protected final TablestoreFacade tablestoreFacade;
+    protected final TablestoreSplit split;
+    protected final TablestoreTableHandle tableHandle;
+    protected final boolean enableLooseCast;
+
+    protected boolean closed;
+    protected int rows;
+    protected long apiNano;
+    protected long readNano;
+
+    public TablestoreAbstractRecordCursor(TablestoreFacade tablestoreFacade, ConnectorSession session, TablestoreSplit split,
```

Comment:
all parameters in one line, or each in a separate line

Reply:
done